### PR TITLE
Add -rdynamic to CMAKE_CXX_FLAGS

### PR DESCRIPTION
**Pull request by aaron-sierra**
Make sure that exported functions (i.e. {device}_input_new()) will land in the dynamic symbol table, regardless of level of optimization.

I experienced missing symbols using GCC 10.3.1 with Buildroot. This option has been available since GCC 4.1.0.

### CODE REVIEW

**Comment by szpajder** [comment]
Doesn't cmake add this flag automatically for you?

Mine does. I'm on cmake 3.23.3.

**Comment by aaron-sierra** [comment]
> Doesn't cmake add this flag automatically for you?

I made my builds verbose to find out.

```
cmake -B build/ -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
```

It does for me, too, when building under Ubuntu 20.04 (cmake 3.16.3). Though that may come from _/usr/share/cmake-{cmake-version}/Modules/Platform/Linux-GNU.cmake_, which is not automatically applied under Buildroot 2022.11-rc2 (cmake 3.22).

```
# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
# file Copyright.txt or https://cmake.org/licensing for details.


# This module is shared by multiple languages; use include blocker.
if(__LINUX_COMPILER_GNU)
  return()
endif()
set(__LINUX_COMPILER_GNU 1)

macro(__linux_compiler_gnu lang)
  # We pass this for historical reasons.  Projects may have
  # executables that use dlopen but do not set ENABLE_EXPORTS.
  set(CMAKE_SHARED_LIBRARY_LINK_${lang}_FLAGS "-rdynamic")
endmacro()
```

## Ubuntu 20.04 - Unpatched v4.0.2
TLDR: Adds `-rdynamic`, when building the final binary resulting in `rtlsdr_input_new()` in the dynamic symbol table.
```shell
/usr/bin/c++   -Wall -Wextra -O3 -DNDEBUG  -rdynamic CMakeFiles/rtl_airband_base.dir/config.cpp.o CMakeFiles/rtl_airband_base.dir/input-common.cpp.o CMakeFiles/rtl_airband_base.dir/input-file.cpp.o CMakeFiles/rtl_airband_base.dir/input-helpers.cpp.o CMakeFiles/rtl_airband_base.dir/mixer.cpp.o CMakeFiles/rtl_airband_base.dir/output.cpp.o CMakeFiles/rtl_airband_base.dir/rtl_airband.cpp.o CMakeFiles/rtl_airband_base.dir/squelch.cpp.o CMakeFiles/rtl_airband_base.dir/util.cpp.o CMakeFiles/rtl_airband_base.dir/udp_stream.cpp.o CMakeFiles/rtl_airband_base.dir/version.cpp.o CMakeFiles/rtl_airband_base.dir/input-rtlsdr.cpp.o  -o rtl_airband   -L/home/user/src/libconfig-issues/libconfig-1.7.3/root/lib  -Wl,-rpath,/home/user/src/libconfig-issues/libconfig-1.7.3/root/lib: -ldl -lm -lpthread -lconfig++ -lmp3lame -lshout -lrtlsdr -lfftw3f
```

## Buildroot 2022.11-rc2 - Unpatched v4.0.2
TLDR: `rtlsdr_input_new()` is *not* in the dynamic symbol table.
```shell
/home/user/src/buildroot/output/host/bin/aarch64-none-linux-gnu-g++ --sysroot=/home/user/src/buildroot/output/host/aarch64-buildroot-linux-gnu/sysroot -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Os -g0 -D_FORTIFY_SOURCE=1 -Wall -Wextra  -DNDEBUG CMakeFiles/rtl_airband_base.dir/config.cpp.o CMakeFiles/rtl_airband_base.dir/input-common.cpp.o CMakeFiles/rtl_airband_base.dir/input-file.cpp.o CMakeFiles/rtl_airband_base.dir/input-helpers.cpp.o CMakeFiles/rtl_airband_base.dir/mixer.cpp.o CMakeFiles/rtl_airband_base.dir/output.cpp.o CMakeFiles/rtl_airband_base.dir/rtl_airband.cpp.o CMakeFiles/rtl_airband_base.dir/squelch.cpp.o CMakeFiles/rtl_airband_base.dir/util.cpp.o CMakeFiles/rtl_airband_base.dir/udp_stream.cpp.o CMakeFiles/rtl_airband_base.dir/version.cpp.o CMakeFiles/rtl_airband_base.dir/input-rtlsdr.cpp.o -o rtl_airband  -ldl -lm -lpthread -lconfig++ -lmp3lame -lshout -lrtlsdr -lfftw3f
```

## Buildroot 2022.11-rc2 - Patched v4.0.2
TLDR: Adds `-rdynamic`, when compiling every object resulting in `rtlsdr_input_new()` in the dynamic symbol table.
```shell
/home/user/src/buildroot/output/host/bin/aarch64-none-linux-gnu-g++ --sysroot=/home/user/src/buildroot/output/host/aarch64-buildroot-linux-gnu/sysroot -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Os -g0 -D_FORTIFY_SOURCE=1 -rdynamic -Wall -Wextra  -DNDEBUG CMakeFiles/rtl_airband_base.dir/config.cpp.o CMakeFiles/rtl_airband_base.dir/input-common.cpp.o CMakeFiles/rtl_airband_base.dir/input-file.cpp.o CMakeFiles/rtl_airband_base.dir/input-helpers.cpp.o CMakeFiles/rtl_airband_base.dir/mixer.cpp.o CMakeFiles/rtl_airband_base.dir/output.cpp.o CMakeFiles/rtl_airband_base.dir/rtl_airband.cpp.o CMakeFiles/rtl_airband_base.dir/squelch.cpp.o CMakeFiles/rtl_airband_base.dir/util.cpp.o CMakeFiles/rtl_airband_base.dir/udp_stream.cpp.o CMakeFiles/rtl_airband_base.dir/version.cpp.o CMakeFiles/rtl_airband_base.dir/input-rtlsdr.cpp.o -o rtl_airband  -ldl -lm -lpthread -lconfig++ -lmp3lame -lshout -lrtlsdr -lfftw3f
```

## Buildroot 2022.11-rc2 - Unpatched v4.0.2 w/ CMAKE_ENABLE_EXPORTS
TLDR: Adds `-Wl,--export-dynamic`, when building the final binary resulting in `rtlsdr_input_new()` in the dynamic symbol table.
```shell
/home/user/src/buildroot/output/host/bin/aarch64-none-linux-gnu-g++ --sysroot=/home/user/src/buildroot/output/host/aarch64-buildroot-linux-gnu/sysroot -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Os -g0 -D_FORTIFY_SOURCE=1 -Wall -Wextra  -DNDEBUG -Wl,--export-dynamic CMakeFiles/rtl_airband_base.dir/config.cpp.o CMakeFiles/rtl_airband_base.dir/input-common.cpp.o CMakeFiles/rtl_airband_base.dir/input-file.cpp.o CMakeFiles/rtl_airband_base.dir/input-helpers.cpp.o CMakeFiles/rtl_airband_base.dir/mixer.cpp.o CMakeFiles/rtl_airband_base.dir/output.cpp.o CMakeFiles/rtl_airband_base.dir/rtl_airband.cpp.o CMakeFiles/rtl_airband_base.dir/squelch.cpp.o CMakeFiles/rtl_airband_base.dir/util.cpp.o CMakeFiles/rtl_airband_base.dir/udp_stream.cpp.o CMakeFiles/rtl_airband_base.dir/version.cpp.o CMakeFiles/rtl_airband_base.dir/input-rtlsdr.cpp.o -o rtl_airband  -ldl -lm -lpthread -lconfig++ -lmp3lame -lshout -lrtlsdr -lfftw3f
```

**Comment by aaron-sierra** [comment]
Perhaps, my patch should add `CMAKE_ENABLE_EXPORTS` to *CMakeLists.txt*?

**Comment by szpajder** [comment]
In this case

```
set_property(TARGET rtl_airband PROPERTY ENABLE_EXPORTS 1)
```

should suffice. Please test.

Hardcoding compiler-specific flags in CMakeFile isn't right. For example, clang does not have `-rdynamic`.

Also please resubmit your PR against the unstable branch. This is where development takes place. master is for stable releases only.

**Comment by aaron-sierra** [comment]
Thanks, for the review. I'll create a new pull request when it's ready.
